tests/functionaltests/Tests/UIKitTests/UIButtonTests.mm

```diff
@@ -36,6 +36,72 @@
     ASSERT_TRUE([backingElement isKindOfClass:[WXFrameworkElement class]]);
 }
 
+@interface UIDeallocTestButton : UIButton {
+    NSCondition* _condition;
+    BOOL* _signaled;
+}
+@end
+
+@implementation UIDeallocTestButton
+
+- (void)setDeallocEvent:(NSCondition*)condition signaledFlag:(BOOL*)signaled {
+    _condition = condition;
+    _signaled = signaled;
+}
+
+-(void)dealloc {
+    [_condition lock];
+    *_signaled = YES;
+    [_condition signal];
+    [_condition unlock];
```

Comment:
sweet; that will be much cleaner

---
In reply to: [98299508](https://github.com/Microsoft/WinObjC/pull/1861#discussion_r98299508) [](ancestors = 98299508)